scheduler/scheduler.go

```diff
@@ -187,6 +215,9 @@ func BuildDependencyGraph(c client.Interface, sel labels.Selector) (DependencyGr
 			depGraph[d.Child].Requires, depGraph[parent])
 		depGraph[parent].RequiredBy = append(
 			depGraph[parent].RequiredBy, depGraph[child])
+		if err := depGraph[parent].updateMeta(d.Meta); err != nil {
```

Comment:
This code ensures that `parent` has sum of metadata from all resources which depends on it, and that in each definition of it's meta - there are same values for each key, or there is no value for previously defined key.

Moving meta into resource should be an easy task, but will require updating its interface (adding SetMeta - noop for other than replicaset resources) which then could be used by Status in ReplicaSet.